### DISCUSSION TITLE
Add the ability to change the color/icon of nodes in the Scene tree

### DISCUSSION THREAD

**Discussion by Fartboy3000** [godotengine/godot-proposals#10112] [Editor]
### Describe the project you are working on

A generic 2D project.

### Describe the problem or limitation you are having in your project

I have a big scene tree, and sometimes I have trouble with spotting/differentiating between the different nodes.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation 

Being able to give the nodes in a tree a different color/icon would greatly improve readability of a tree, making it possible to differentiate between nodes at the glance of an eye.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Just like assigning a different color to folders in the FileSystem, a user could right-click on a node in the tree and give it a different color/icon.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I do not have any experience with changing how Godot works internally, I do not know.]

### Is there a reason why this should be core and not an add-on in the asset library?

Just like assigning colors to folders in the FileSystem, assigning colors to nodes should be a core feature.

### Additional notes

If assigning different icons to nodes is a bad idea, I think being able to change the color of a node icon would already greatly improve readability.

**Comment by AThousandShips** [comment]
Thank you for your proposal, consolidating in:
* https://github.com/godotengine/godot-proposals/issues/8118
